### PR TITLE
Add support for custom handler IDs

### PR DESCRIPTION
**Pull request by D-Cysteine**
Currently, NEI uses the class name as the handler ID. I'm working on adding some new NEI handlers, but they all use the same class, so they collide.

This PR adds a (hopefully) backwards-compatible way to manually set the handler ID, rather than always using the Java class name. I've loaded up the game and checked a bunch of handlers, and they all seem to work, but there are too many to test thoroughly.

@TimeConqueror, this PR modifies the signature of `NEIRegisterHandlerInfosEvent`, so please take a look at that. And thanks for adding that event!

### CODE REVIEW

File: src/codechicken/nei/recipe/HandlerInfo.java [hunk 1]
```diff
@@ -160,8 +160,8 @@ public void setYShift(int yShift) {
     public static class Builder {
         private final HandlerInfo info;
 
-        public Builder(Class<?> handlerClazz, String modName, String modId) {
-            this.info = new HandlerInfo(handlerClazz.getName(), modName, modId, true, null);
+        public Builder(String handlerName, String modName, String modId) {
```
Comment (Glease) [inline src/codechicken/nei/recipe/HandlerInfo.java:163]: Can probably just create an overload. 100% backwards compatible.

File: src/codechicken/nei/event/NEIRegisterHandlerInfosEvent.java [hunk 2]
```diff
@@ -20,8 +19,8 @@
  * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
  */
 public class NEIRegisterHandlerInfosEvent extends Event {
-    public void registerHandlerInfo(Class<? extends IRecipeHandler> handlerClazz, String modName, String modId, Consumer<HandlerInfo.Builder> builder) {
-        HandlerInfo.Builder b = new HandlerInfo.Builder(handlerClazz, modName, modId);
+    public void registerHandlerInfo(String handlerName, String modName, String modId, Consumer<HandlerInfo.Builder> builder) {
```
Comment (Glease) [inline src/codechicken/nei/event/NEIRegisterHandlerInfosEvent.java:22]: Can probably just create an overload. 100% backwards compatible.

**Review by Glease — CHANGES_REQUESTED** [review]
Adding overloads and delegates the actual business logic (and adding `@Deprecated` to old APIs if necessary) is a common pattern when evolving existing APIs. It allows adding new functionality while maintaining binary and source code compatibility.

File: src/codechicken/nei/recipe/HandlerInfo.java (hunk 1 above)
Comment (D-Cysteine) [reply src/codechicken/nei/recipe/HandlerInfo.java:163]: Thanks! Added

File: src/codechicken/nei/event/NEIRegisterHandlerInfosEvent.java (hunk 2 above)
Comment (D-Cysteine) [reply src/codechicken/nei/event/NEIRegisterHandlerInfosEvent.java:22]: Thanks! Added

**Review by Glease — APPROVED** [review]

**Pull request merged**